awx/main/models/credential/__init__.py

```diff
@@ -452,6 +452,9 @@ def get_input(self, field_name, **kwargs):
             return self.inputs[field_name]
         if 'default' in kwargs:
             return kwargs['default']
+        for field in self.credential_type.inputs.get('fields', []):
```

Comment:
@AlanCoding @jakemcdermott how do you feel about this?  Up until now, we've not really had an notion of the default fallback actually _coming_ from the schema, because we've never really _needed_ it - all of our booleans up to this point have defaulted to `False`.

Reply:
I would like a section under "Defining Custom Credential Type Inputs"

https://github.com/ansible/awx/blob/devel/docs/custom_credential_types.md

For "Field required and default behavior"

If a field is not required, and there is no default, then we're kind of in hot water, right?

I do agree what you're saying and what you're doing here in an immediate sense, but I fumbled around these conversations in Jake's PR a lot.